main.py

```diff
@@ -1 +1,183 @@
-#This is the main file 
+from flask import Flask, request, render_template, redirect
+import flask
+from qrgen import GPQRGen
+from airtable import Airtable
+import json
+import requests
+
+#Getting config stuff
+with open("config.json") as f:
+    config = json.load(f);
+    airtable_auth_key = config["airtable_auth_key"]
+    base_key = config["base_key"] 
+    client_id = config["client_id"]
+    clientSecret = config["client_secret"]
+    secretKey = config["secret_key"]
+    botUserId = config["softbank_user_id"]
+    snekCoin_API_Key = config["snekCoin_API_Key"]
+
+app = Flask(__name__)
+app.secret_key = secretKey
+
+#Airtable bases
+claims_base = Airtable(base_key, "Coin Claims", api_key=airtable_auth_key)
+admin_base = Airtable(base_key, "Verified Generators", api_key=airtable_auth_key)
+
+@app.route("/", methods=["GET", "POST"])
+def index():
+    return ("This is the index.", 200)
+
+@app.route("/generate", methods=["GET", "POST"])
+def generate():
+    #Making sure that admin has logged in before
+    try:
+        admin_record = admin_base.search("Slack Tag", flask.session["authdict"]["name"])
+        logged_in = admin_record[0]["fields"]["Logged In?"]
+        username = admin_record[0]["fields"]["Slack Tag"]
+    except KeyError:
+        logged_in = False
+        
+    if request.method == 'POST':
+        amount = request.form.get("amount")
+        flask.session["amount"] = int(amount)
+        return redirect("/qrcode")
+    
+    if not logged_in:
+        return render_template("adminlogin.html", clientId=client_id)
+    else:
+        return render_template("generate.html", name=username)
+    
+@app.route("/qrcode", methods=["GET", "POST"])
+def qrcode():
+    #Generates the QR Code
+    gen = GPQRGen("http://localhost:3000", flask.session["amount"])

```

Comment:
can you make this `0.0.0.0` instead of localhost? that will bind to all interfaces

Reply:
0.0.0.0 never works for me when i try to test using it and i have no idea why. Any ideas?

Reply:
I'm not sure whether `0.0.0.0` works on windows like it does on linux -
might be best to change it during testing? We need this for production
though so it can run properly within the docker container.